webview/src/setup/state/setupDataSlice.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import {
+  DEFAULT_SECTION_COLLAPSED,
+  SectionCollapsed,
+  SetupData
+} from 'dvc/src/setup/webview/contract'
+
+export interface SetupDataState extends SetupData {
+  hasReceivedMessageFromVsCode: boolean
+}
+
+export const setupDataInitialState: SetupDataState = {
+  canGitInitialize: false,
+  cliCompatible: undefined,
+  dvcCliDetails: undefined,
+  hasData: undefined,
+  hasReceivedMessageFromVsCode: false,
+  isPythonExtensionUsed: false,
+  isStudioConnected: false,
+  needsGitCommit: false,
+  needsGitInitialized: false,
+  projectInitialized: false,
+  pythonBinPath: undefined,
+  sectionCollapsed: DEFAULT_SECTION_COLLAPSED,
+  shareLiveToStudio: false
```

Comment:
Should we break up this slice into sections? Aka, `studio`, `dvc`, `experiments`, and `sectionCollapsed`.

Reply:
Decided to do so :)